src/components/TodoCard.tsx

```diff
@@ -0,0 +1,104 @@
+import { Todo } from '../types/typedefs';
+import { useTodos } from '../hooks/useTodos';
+import classNames from 'classnames';
+import { deleteTodo, updateTodo } from '../api/todosMethods';
+
+interface TodoCardProps {
+  todoListState: ReturnType<typeof useTodos>;
+  todo: Todo;
+  loadingTodoId: number | null;
+  setLoadingTodoId: (id: number | null) => void;
+}
+
+export const TodoCard: React.FC<TodoCardProps> = ({
+  todoListState,
```

Comment:
you can't pass all state
imagine how many rerenders you have
pass only what you need